lib/tokenlist.cpp

```diff
@@ -1768,6 +1768,10 @@ void TokenList::validateAst() const
             tok = tok->link();
             continue;
         }
+        if (Token::simpleMatch(tok, ". decltype (") && tok->linkAt(2)) { // skip trailing return type
```

Comment:
hmm.. that `tok->linkAt(2)` really should be non-null for a "(" token.. if there is not matching parentheses cppcheck fails early.